### PR TITLE
Gathering state cleanup

### PR DESCRIPTION
**Pull request by pthatcherg**
Also clarify when regathering can occur (after an ICE restart).

Attempts to fix https://github.com/w3c/webrtc-pc/issues/808 and
https://github.com/w3c/webrtc-pc/issues/812

### CODE REVIEW

**Comment by pthatcherg** [comment]
FYI, I accidentally based on https://github.com/w3c/webrtc-pc/pull/815, so look at the diff between it and this.  I can fix this PR if we need.  Or we can land https://github.com/w3c/webrtc-pc/pull/815 :).

File: webrtc.html [hunk 1]
```diff
@@ -6245,7 +6252,7 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
     readonly        attribute RTCIceRole           role;
     readonly        attribute RTCIceComponent      component;
     readonly        attribute RTCIceTransportState state;
-    readonly        attribute RTCIceGatheringState gatheringState;
+    readonly        attribute RTCIceGathererState  gatheringState;
```
Comment (adam-be) [inline webrtc.html]: Could RTCIceGathererState be RTCIceTransportGatheringState to keep the word "gathering", but be explicit that it belongs to the RTCIceTransport?

Comment (adam-be) [inline webrtc.html]: Or rename the state type on RTCPeerConnection to RTCAggregatedIceGatheringState

**Comment by alvestrand** [comment]
Names were chosen to be consistent with ORTC.

**Comment by alvestrand** [comment]
Seems merge-ready once Peter's repo has Bernard's fix.

**Comment by aboba** [comment]
@pthatcherg  Would be nice to merge this PR for the next Editor's draft.  Can you merge the wording tweak that Harald suggested?

**Comment by stefhak** [comment]
@pthatcherg can you take look.

**Comment by alvestrand** [comment]
Rebased in #882

**Pull request closed without merging**